modules/time-shortcode/load.php

```diff
@@ -10,114 +10,150 @@
 
-	// add the time converter JS code
-	add_action( 'wp_footer', 'o2_time_converter_script' );
+		global $shortcode_tags, $in_comment_content;
+		$in_comment_content = true;
 
-	// return the new link
-	return $out;
-}
+		// save the shortcodes
+		$original_tags = $shortcode_tags;
 
-function o2_time_converter_script() {
-	global $wp_locale; // These are strings that Core has already translated.
-?>
-	<script type="text/javascript">
-	( function( $ ) {
-		var ts = <?php echo json_encode( array(
-			'months' => array_values( $wp_locale->month ),
-			'days'   => array_values( $wp_locale->weekday ),
-		) ); ?>;
-
-		var o2_parse_date = function ( text ) {
-			var m = /^([0-9]{4})-([0-9]{2})-([0-9]{2})T([0-9]{2}):([0-9]{2}):([0-9]{2})\+00:00$/.exec( text );
-			return new Date( Date.UTC( +m[1], +m[2] - 1, +m[3], +m[4], +m[5], +m[6] ) );
-		}
-		var o2_format_date = function ( d ) {
-			var p = function( n ) {
-				return ( '00' + n ).slice( -2 );
-			};
-			var tz = -d.getTimezoneOffset() / 60;
-			if ( tz >= 0 ) {
-				tz = "+" + tz;
+		// only process the time shortcode
+		$shortcode_tags = array( 'time' => array( 'o2_Time_Shortcode', 'time_shortcode' ) );
+
+		// do the time shortcode on the comment
+		$comment_text = do_shortcode( $comment_text );
+
+		// restore the normal shortcodes
+		$shortcode_tags = $original_tags;
+
+		$in_comment_content = false;
+		return $comment_text;
+	}
+
+	/**
+	 * Render and enqueue the javascript that converts the raw time into local time
+	 **/
+	public static function time_conversion_script() {
+
+		global $wp_locale; // These are strings that Core has already translated.
+
+		?>
+		<script type="text/javascript">
+
+			var o2_get_time_settings = function(){
+				return <?php echo json_encode( array(
+					'months' => array_values( $wp_locale->month ),
+					'days'   => array_values( $wp_locale->weekday ),
+				) ); ?>;
 			}
-			return "" + ts['days'][ d.getDay() ] + ", " + ts['months'][ d.getMonth() ] + " " + p( d.getDate() ) + ", " + d.getFullYear() + " " + p( d.getHours() ) + ":" + p( d.getMinutes() ) + " UTC" + tz;
-		}
 
-		$( 'body' ).on( 'ready post-load ready.o2', function() {
-			$( 'abbr.globalized-date' ).each( function() {
-				t = $( this );
-				var d = o2_parse_date( t.attr( 'title' ) );
-				if ( d ) {
-					t.text( o2_format_date( d ) );
-				}
-			} );
-		} );
-	} )( jQuery );
-	</script>
-<?php
-}
-add_shortcode( 'time', 'o2_time_shortcode' );
+			jQuery( 'body' ).on( 'ready post-load ready.o2', function() {
 
-function o2_time_shortcode_in_comments( $comment ) {
-	global $shortcode_tags, $post, $time_in_comments;
-	$time_in_comments = true;
-	$post = get_post();
+				jQuery( 'abbr.globalized-date' ).each( function() {
 
-	// save the shortcodes
-	$saved_tags = $shortcode_tags;
+					var $this = jQuery( this );
 
-	// only process the time shortcode
-	$shortcode_tags = array( 'time' => 'o2_time_shortcode' );
+					var time = $this.data( 'time' );
```

Comment:
I found this surprising that `jQuery.data()` returns a Number (rather than a String) here. Can we make that clear here? A code comment? Sending it through `Number()`? Something else?